corehq/apps/app_manager/templates/app_manager/v2/import_app.html

```diff
@@ -1,40 +1,24 @@
 {% extends "app_manager/v2/apps_base.html" %}
 {% load i18n %}
 {% load hq_shared_tags %}
+{% load compress %}
 
-{% block js-inline %}{{ block.super }}
-    <script src="{% static 'hqwebapp/js/lib/compression.js' %}"></script>
-    <script>
-        $(function () {
-            $(".historyBack").click(function () {
-                history.back();
-                return false;
-            });
-        });
-
-        function CompressionViewModel(source, post_url){
-            var self = this;
-            self.name = ko.observable("");
-            self.source = ko.observable(source);
-            self.save = function(formElement) {
-                $.postGo(post_url,
-                         {name : self.name(),
-                          compressed: LZW.compress(self.source())}
-                        );
-                return false;
-            };
-        }
-        var source = {{ app.export_json|JSON }};
-        var post_url = "{% url "import_app" domain %}";
-        $('#app-import-form').koApplyBindings(new CompressionViewModel(source, post_url));
-    </script>
+{% block js %}{{ block.super }}
+    {% compress js %}
+        <script src="{% static 'hqwebapp/js/initial_page_data.js' %}"></script>
+        <script src="{% static 'hqwebapp/js/urllib.js' %}"></script>
+        <script src="{% static 'hqwebapp/js/lib/compression.js' %}"></script>
+        <script src="{% static 'app_manager/js/import_app.js' %}"></script>
+    {% endcompress %}
 {% endblock %}
 
 {% block breadcrumbs %}
     <li>{% trans 'Import Application' %}</li>
 {% endblock %}
 
 {% block page_content %}
+    {% registerurl 'import_app' domain %}
+    {% initial_page_data 'export_json' app.export_json %}
```

Comment:
Possibly...I'm not really sure what the use case is where `app` has a value. It looks like there's sometimes an app or a different domain provided via GET (looking at https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/app_manager/views/apps.py#L406-L407) but I haven't found code that triggers that workflow. Basically stopped investigating for fear that I was getting into a rabbit hole.